### ISSUE TITLE
Odd crash on Expo when minified and in production mode

### ISSUE THREAD

**Issue by trajano** [reduxjs/redux-toolkit#2936]
### Prior Issues

* https://stackoverflow.com/questions/73788057/react-native-project-crashing-with-redux-toolkit
* https://stackoverflow.com/questions/74283724/expo-crashes-with-redux-toolkit
* Also have a long discussion here https://forums.expo.dev/t/expo-go-with-react-redux-crashes-on-published-app-but-works-in-local/67949/8

### What is the current behavior?

The app crashes immediately.  Note this does not occur if the app is not minimized or if in non-production.  

### Steps to Reproduce

I am unable to make a MVCE to show this.  

But this can be recreated in my app https://github.com/trajano/spring-cloud-demo (it's not a trivial app, but to make it crash I when I add the following to https://github.com/trajano/spring-cloud-demo/blob/rework/expo-app/authenticated-context/AuthenticatedProvider.tsx

```
import { configureStore } from "@reduxjs/toolkit";
export const store = configureStore({
  reducer: {},
});
```

### What is the expected behavior?

Not crash.

### Environment Details

    "@reduxjs/toolkit": "^1.9.0",
    "react-redux": "^8.0.5",
    "redux-persist": "^6.0.0",
    "expo": "^47.0.6",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-native": "0.70.5",

**Comment by markerikson** [comment]
I'm kind of confused. The check you're pointing to is in the `redux` core `createStore` function, but this only happens if you use RTK's `configureStore`, and works fine with just `createStore` by itself?

**Comment by markerikson** [comment]
@timdorr @phryneas so tracing back the history of those lines, it dates back to this issue/PR in late 2015:

- https://github.com/reduxjs/redux/issues/1029
- https://github.com/reduxjs/redux/pull/1075

Worst case I think we'd be okay to remove that warning?

**Comment by timdorr** [comment]
I know I'm the one that originally merged this, but I think it's probably better to remove it. As we move towards ESM being more and more popular, relying on a Node-specific global is going to become less likely to be present and increase these kinds of issues. 

At least, that's my general sense of things. I have no data to back that up, so just take it as a hunch.

**Comment by markerikson** [comment]
Eh, I'm less worried about the `NODE_ENV` thing than I am the "does this 'non-minified code' warning even still make sense in 2022?" question :)

**Comment by trajano** [comment]
> I'm kind of confused. The check you're pointing to is in the `redux` core `createStore` function, but this only happens if you use RTK's `configureStore`, and works fine with just `createStore` by itself?

The check I am poiting at is in `lib/redux.js` or `redux`.   The toolkit does `import * from redux` so I was trying to hunt down what could cause it.   Like I said the behaviour is sort of weird at least let me see a JS crash trace or allow me to wrap in a try-catch and log the error, but so far  and I haven't been able to pull it out on its own.

**Comment by markerikson** [comment]
@trajano : yes, I know, that's why I'm confused why you're saying this happens with RTK's `configureStore()`, but not just the core `createStore()`.  That code is in the module-level `redux/src/index.js` file, so it runs as soon you import anything from `redux`.  Therefore, this error shouldn't depend on whether or not you're using RTK.

**Comment by trajano** [comment]
Well a similar error occurs on RTK (just crash) but not l don't know what exactly is causing it.  I am just pointing out that the patch I did on redux.js seems to fix createStore in redux.  Maybe there's similar code in RTK, but by the time I can "see" it, the code is minified to an unreadable state already.

**Comment by markerikson** [comment]
Yeah, I understand that you're seeing this crash when you call `configureStore()`.

But because this code is in the `redux` core, I'm asking:

**does this _also_ crash if you just use `const store = createStore()`?**

because the "RTK" part _shouldn't_ matter here.

**Comment by trajano** [comment]
> **does this _also_ crash if you just use `const store = createStore()`?**

You mean 
```
import { legacy_createStore as createStore } from "redux";
export const store = createStore((state = 0, action) => state);
```

Yes it crashes unless I remove that `isCrushed` check.  So if that check was removed at least the "legacy" version of Redux would still work.

**Comment by phryneas** [comment]
If that check was removed, all versions would - this is really just the core implementation.

**Comment by trajano** [comment]
> If that check was removed, all versions would - this is really just the core implementation.

you mean it would cause crashes on other places if that `isCrushed` check was removed @phyrneas?

**Comment by markerikson** [comment]
No. We're saying that **there is only one place that check exists, in the `redux` core library `index.js` file**.  So it doesn't matter if you're using "legacy Redux" (the `redux` package), or Redux Toolkit (which _depends_ on the `redux` package) - removing that check from `redux` would fix this everywhere `redux` is used.
